tiledb/sm/subarray/subarray.h

```diff
@@ -580,27 +600,24 @@ class Subarray {
       const std::vector<T>& tile_coords, std::vector<uint8_t>* aux) const;
 
   /**
-   * Returns the tile overlap of the subarray.
-   *
-   * The outer vector is indexed by fragment ids and the inner vector is
-   * indexed by range indexes.
+   * Returns the internal `TileOverlap` instance. The caller is responsible
+   * for ensuring that the input indexes are valid.
    *
-   * As an optimization, the underlying data structure may be shared between
-   * `Subarray` instances and their partitioned `Subarray` instances created
-   * by the `SubarrayPartitioner`. The caller must use the
-   * `Subarray::overlap_range_offset` to determine the range index that points
-   * to the starting range of this `Subarray` instance.
+   * @param fragment_idx The fragment index.
+   * @param range_idx The range index.
    */
-  const std::vector<std::vector<TileOverlap>>& tile_overlap() const;
+  inline const TileOverlap* get_tile_overlap(
+      uint64_t fragment_idx, uint64_t range_idx) const {
+    return tile_overlap_.get(fragment_idx, range_idx);
+  }
 
   /**
-   * The `Subarray::tile_overlap` returns a data structure where the
-   * outter vector is indexed by fragment ids and the inner vector
-   * is indexed by range ids. This API returns the offset into the
-   * inner vector that points to the first range used by this
-   * `Subarray` instance.
+   * Returns the precomputed tile overlap. This may not contain
+   * tile overlap information for each range in this subarray. The
+   * caller is responsible for inspecting the returned object to
+   * determine the ranges it has information for.
    */
-  uint64_t overlap_range_offset() const;
+  const SubarrayTileOverlap* get_subarray_tile_overlap() const;
```

Comment:
Rename to `subarray_tile_overlap`? We use `get_*` if we get an object by passing a pointer as an argument instead of getting it from the return value.